padl/dumptools/inspector.py

```diff
@@ -352,70 +276,51 @@ def get_segment_from_frame(caller_frame: types.FrameType, segment_type, return_l
         full_source = get_source(caller_frame.f_code.co_filename)
 
     lineno = caller_frame.f_lineno
+
     # in python <= 3.7, the lineno points to the end of
     # the statement rather than the beginning, therefore we need to decrement it
     if segment_type == 'call':
         lines = original(full_source).split('\n')
         while '(' not in lines[lineno - 1] and not lines[lineno - 1].startswith('@'):
             lineno -= 1
 
-    source, offset = get_statement(original(full_source), lineno)
-    # the source can contain surrounding stuff we need to discard
-    # as we only have the line number (this is what makes this complicated)
-
-    # get all segments in the source that correspond to calls and might thus
-    # potentially be the class init
-    candidate_segments = var2mod.Finder(node_type).get_source_segments(source)
-    # for each candidate, disassemble and compare the instructions to what we
-    # actually have, a match means this is the correct statement
-    if not candidate_segments:
-        raise RuntimeError(f'{segment_type} not found.')
     # disassemble and get the instructions up to the current position
     target_instrs = _instructions_up_to_offset(caller_frame.f_code,
                                                caller_frame.f_lasti)
-
     # filter out EXTENDED_ARG (instruction used for very large args in target_instrs, this won't
     # be present in instrs)
     target_instrs = [x for x in target_instrs if x.opname != 'EXTENDED_ARG']
 
-    segment = None
-    locs = None
-    found = False
-
-    for segment, locs in sorted(candidate_segments, key=lambda x: -len(x[0])):
-
+    def check_segment(segment):
```

Comment:
Could we make this a function outside of the scope of `get_segment_from_frame`? I think it will make it easier to read as a new developer.

Reply:
Not sure I agree, can you explain why you think it would make it easier to read? I think there is some benefit in having it close to the one place where it's being used.

Reply:
I see. Yes I agree that having it closer to where it's used can be helpful. For me it becomes easier to see the dependencies of a function if all external variables used inside the function are passed to it through the function call. Similar to how you built `instructions_are_the_same`.

Reply:
okay sounds good